frontend/src/liboperator/operator/operator.go

```diff
@@ -66,7 +74,28 @@ func SetupDeviceEndpoint(pool *DevicePool, config apiv1.InfraConfig, path string
 			}
 			go deviceEndpoint(NewJSONUnix(c), pool, config)
 		}
+	}, nil
+}
+
+// Sets up a unix socket for server control and returns a function that listens on the socket
+// until an error occurrs.
+func SetupPreregistrationEndpoint(pool *DevicePool, path string) (func() error, error) {
```

Comment:
nit: should this be `SetupControlEndpoint` instead?

Reply:
done